sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py

```diff
@@ -0,0 +1,112 @@
+from noodles_fasta_wrapper import IndexedFastaReader
+
+class SequenceAccessor:
+    # NOTE: we could totally handle this stuff in Rust if we want.
+    def __init__(self, reader, seqid, length):
+        self.reader = reader
+        self.seqid = seqid
+        self.length = length
+
+    def __getitem__(self, key):
+        if isinstance(key, slice):
+            # Handle slice for range queries
+            start = key.start if key.start is not None else 0
+            stop = key.stop if key.stop is not None else self.length
+
+            # Bounds checking for slice
+            if start < 0 or stop > self.length:
+                raise IndexError(f"Range [{start}:{stop}] is out of bounds for '{self.seqid}' with length {self.length}.")
+
+            region_str = f"{self.seqid}:{start + 1}-{stop}"  # +1 for 1-based indexing
+            return self.reader.query_region(region_str)
+
+        elif isinstance(key, int):
+            # Handle single integer for single nucleotide queries
+            if key < 0 or key >= self.length:
+                raise IndexError(f"Position {key} is out of bounds for '{self.seqid}' with length {self.length}.")
+
+            # Query single nucleotide by creating a 1-length region
+            region_str = f"{self.seqid}:{key + 1}-{key + 1}"  # +1 for 1-based indexing
```

Comment:
You have a test that covers this, right?  This kind of thing is so easy to get wrong. I'd like to see a test covering slices and single index lookups.

Reply:
Seems right though, to be clear! I just don't trust myself with this kind of thing. Also someone in the future could easily break it and not realize.

Reply:
definitely needs tests, all of it haha.